OsmAnd/src/net/osmand/plus/helpers/WaypointHelper.java

```diff
@@ -464,6 +464,8 @@ public void announceVisibleLocations() {
 										if (atd.isTurnStateActive(atdSpeed, dist, STATE_SHORT_ALARM_ANNOUNCE)) {
 											locationPointsStates.put(point, ANNOUNCED_DONE);
 											proceed = false;
+											//update lastAlarm location for avoid series of same types alarms
```

Comment:
comment is not needed, location should be updated once it's in the list ? AlarmInfo lastAlarm = lastAnnouncedAlarms.get(t);